evm/evm-processor/src/ds-rpc/mapping.ts

```diff
@@ -64,15 +65,22 @@ function tryMapBlock(rpcBlock: RpcBlock, req: MappingRequest): Block {
     if (req.transactionList) {
         for (let i = 0; i < transactions.length; i++) {
             let stx = transactions[i]
-            let tx = new Transaction(header, i)
+            let isUnorderedChain = typeof stx == 'string' ? false : (!!stx.chainId && !!stx.transactionIndex && UNORDERED_TRANSACTION_CHAINS.includes(stx.chainId))
+            let index =  typeof stx == 'string' ? i : (isUnorderedChain && !!stx.transactionIndex) ? stx.transactionIndex : i
+            let tx = new Transaction(header, index)
+
             if (typeof stx == 'string') {
                 if (req.fields.transaction?.hash) {
                     tx.hash = stx
                 }
             } else {
                 let {transactionIndex, ...props} = stx
                 Object.assign(tx, props)
-                assert(transactionIndex === i)
```

Comment:
Since `index` is already conditionally set by `isUnorderedChain`, should this just be `assert(transactionIndex === index)`?

Reply:
this is the way they originally checked it for regular evm chains. checking for index or i  would work. I think this way clarifies it's specifically supposed to be in order. but index and i are the same here

Reply:
I'm just wondering why it's not
```
if (isUnorderedChain) {
    assert(transactionIndex === index)
} else {
    assert(transactionIndex === i)
}
```

nor

```
assert(transactionIndex === index)
```

Fine with any way if justified. Just want to know the reasoning.